scipy/stats/_discrete_distns.py

```diff
@@ -504,44 +504,33 @@ def _logpmf(self, k, M, n, N):
         return result
 
     def _pmf(self, k, M, n, N):
-        # same as the following but numerically more precise
-        # return comb(good, k) * comb(bad, N-k) / comb(tot, N)
-        return exp(self._logpmf(k, M, n, N))
+        return _boost._hypergeom_pdf(k, n, N, M)
+
+    def _cdf(self, k, M, n, N):
+        return _boost._hypergeom_cdf(k, n, N, M)
 
     def _stats(self, M, n, N):
-        # tot, good, sample_size = M, n, N
-        # "wikipedia".replace('N', 'M').replace('n', 'N').replace('K', 'n')
-        M, n, N = 1.*M, 1.*n, 1.*N
+        M, n, N = 1. * M, 1. * n, 1. * N
         m = M - n
-        p = n/M
-        mu = N*p
 
-        var = m*n*N*(M - N)*1.0/(M*M*(M-1))
-        g1 = (m - n)*(M-2*N) / (M-2.0) * sqrt((M-1.0) / (m*n*N*(M-N)))
-
-        g2 = M*(M+1) - 6.*N*(M-N) - 6.*n*m
-        g2 *= (M-1)*M*M
-        g2 += 6.*n*N*(M-N)*m*(5.*M-6)
-        g2 /= n * N * (M-N) * m * (M-2.) * (M-3.)
-        return mu, var, g1, g2
+        g2 = M * (M + 1) - 6. * N * (M - N) - 6. * n * m
+        g2 *= (M - 1) * M * M
+        g2 += 6. * n * N * (M - N) * m * (5. * M - 6)
+        g2 /= n * N * (M - N) * m * (M - 2.) * (M - 3.)
+        return (
+            _boost._hypergeom_mean(n, N, M),
+            _boost._hypergeom_variance(n, N, M),
+            _boost._hypergeom_skewness(n, N, M),
+            g2,  # Boost kurtosis_excess doesn't return the same as the value computed here.
```

Comment:
Does it have to do with the difference between kurtosis and excess/Fisher kurtosis, or does it look wrong?

Reply:
If I understand the [wikipedia](The excess kurtosis is defined as kurtosis minus 3. There are 3 distinct regimes as described below. ) correctly, you should just be able to add 3 to get kurtosis from excess kurtosis?

Reply:
I wasn't able to get the same values as expected. Ill dig into it tonight